### PR TITLE
Fixed role based accesses

### PR DESCRIPTION
**Pull request by RodriFS**

### CODE REVIEW

File: src/Pages/ChannelRequests.razor [hunk 1]
```diff
@@ -9,11 +9,18 @@
 
 <PageTitle>Channel Operation Requests</PageTitle>
 
-@if (!_hidePendingRequests)
```
Comment (Jossec101) [inline src/Pages/ChannelRequests.razor:12]: What would be render if there are no requests, an empty datagrid?

File: src/Pages/Wallets.razor [hunk 2]
```diff
@@ -284,8 +283,14 @@
...
     {
-
-        _wallets = await WalletRepository.GetAll();
+        if (ClaimsPrincipal != null && ClaimsPrincipal.IsInRole(ApplicationUserRole.FinanceManager.ToString()))
+        {
+            _wallets = await WalletRepository.GetAll();
+        }
+        else
+        {
+           _wallets = await WalletRepository.GetAvailableWallets();
```
Comment (Jossec101) [inline src/Pages/Wallets.razor:292]: What's the reason behind this if else?

File: src/Pages/Withdrawals.razor [hunk 3]
```diff
@@ -355,59 +352,40 @@
...
-    private IWalletRepository WalletRepository { get; set; }
-
-    [Inject]
-    private IBitcoinService BitcoinService { get; set; }
+@inject IToastService ToastService
+@inject IWalletWithdrawalRequestRepository WalletWithdrawalRequestRepository
+@inject IWalletWithdrawalRequestPsbtRepository WalletWithdrawalRequestPsbtRepository
+@inject IWalletRepository WalletRepository
+@inject IBitcoinService BitcoinService
+@inject ISchedulerFactory SchedulerFactory
```
Comment (Jossec101) [inline src/Pages/Withdrawals.razor:379]: We can do injects in both ways, but it is true that we have not used the same approach always

File: src/Pages/ChannelRequests.razor (hunk 1 above)
Comment (RodriFS) [reply src/Pages/ChannelRequests.razor:12]: For finance managers will render the requests waiting for signature.
For node managers will render the requests the node manager created and are pending signature from the finance manager.
For admins it won't render at all. All requests will show in the other datagrid

The thing is that i needed the "new" button for node managers, but the button needs the datagrid to exist, so i had to separate them into pending and approved for node managers too

File: src/Pages/ChannelRequests.razor [hunk 4]
```diff
@@ -9,11 +9,18 @@
...
+@if (_isFinanceManager || _isNodeManager)
 {
     <Field Flex="Flex.JustifyContent.Between">
+    @if (_isFinanceManager)
+    {
         <h3>Requests awaiting my signature</h3>
+    }
+    else if (_isNodeManager)
+    {
+        <h3>Requests awaiting signatyre by a Finance Manager</h3>
```
Comment (RodriFS) [inline src/Pages/ChannelRequests.razor:21]: Oops, typo. will fix

File: src/Pages/Wallets.razor (hunk 2 above)
Comment (RodriFS) [reply src/Pages/Wallets.razor:292]: For node managers and admins to not see pending wallets. I think is pointless.

File: src/Pages/Withdrawals.razor (hunk 3 above)
Comment (RodriFS) [reply src/Pages/Withdrawals.razor:379]: There were a lot of null warnings on this, that's why I moved it.
I tried to remove all null warnings but some of them are literally impossible to remove. For example when you have a ref in a component

Comment (Jossec101) [reply src/Pages/Withdrawals.razor:379]: Yes null handling cannot be complete removed in C# even with modern features

**Review by Jossec101 — APPROVED** [review]
LGTM

**Pull request merged**